frontend/src/components/NewTodo.tsx

```diff
@@ -1,38 +1,77 @@
-import React, { useRef, useContext } from 'react';
+import React, { useContext, useReducer } from 'react';
 import styles from './styles/NewTodo.module.css';
 import { TodosContext } from '../store/todos-context';
 import axios from 'axios';
-import { TodosContextModel } from './models/TodosContext';
 import environment from '../environments/environment.dev';
+import { TodosContextModel } from './models/TodosContext';
+import { Action, State, StateTodo } from './models/Todo';
 
 const NewTodo : React.FC = () =>
 {
+
 	const todosContext : TodosContextModel = useContext(TodosContext);
-	const todoTextInputRef : React.MutableRefObject<HTMLInputElement> = useRef<HTMLInputElement>(null);
+	const initialState : StateTodo = { title: '', url: '', description: '' };
+	const [ todoInputs, dispatchTodoInputs ] = useReducer(reducerHandler, initialState);
+
+	function reducerHandler(state : State, action : Action)
+	{
+		switch (action.type)
+		{
+		case 'TODO_TITLE':
```

Comment:
Why is there one action for each property? Why not change all at once? the action-name also says nothing what is happening... TODO_TITLE... does it add, delete, update? improve your namings please